src/assemble_workflow/dist.py

```diff
@@ -72,6 +76,97 @@ def __extract__(self, dest):
         with tarfile.open(self.path, "r:gz") as tar:
             tar.extractall(dest)
 
-    def __build__(self, name, dest):
-        with tarfile.open(name, "w:gz") as tar:
+    def __build__(self, bundle_recorder, dest):
+        with tarfile.open(bundle_recorder.package_name, "w:gz") as tar:
             tar.add(self.archive_path, arcname=os.path.basename(self.archive_path))
+
+
+class DistRpm(Dist):
+    def __extract__(self, dest):
+        with tarfile.open(self.path, "r:gz") as tar:
+            tar.extractall(dest)
+
+    def __build__(self, bundle_recorder, dest):
+        logging.info("build for rpm distribution.")
+        manifest_data = bundle_recorder.bundle_manifest.data["build"]
+        product_name = manifest_data["name"].lower()
+        if product_name == "opensearch dashboards":
+            product_name = "-".join(product_name.split())
+            product_name_alt = "_".join(product_name.split())
+        output_dir = os.getcwd()
```

Comment:
Should come from the outside, don't assume current directory.

Reply:
I see. This is not the output directory. os.getcwd() need to be used to get the current working directory which is used  for other variables, like scripts_path, to refer. I will change the name from output_dir to root.

Reply:
If you need paths relative to this file, they can be computed from `__file__`, like in `ScriptFinder`. Generally avoid `chdir` and `getcwd`.